### PR TITLE
chore: Remove new output feature flag [CFG-2019]

### PR DESCRIPTION
**Pull request by ofekatr**
- [x] Follows [CONTRIBUTING](https://github.com/snyk/snyk/blob/master/CONTRIBUTING.md) rules

#### What does this PR do?

- Removes the `iacCliOutputRelease` feature flag.
- Removes the legacy output paths, since they're no longer reachable.

#### How should this be manually tested?
- Run `snyk iac test` and ensure the test results are outputted as expected in all formats.

#### Any background context you want to provide?

- Once the New CLI Output has been in GA for 1 month, we would like to remove the `iacCliOutputRelease` feature flag from the code altogether.

#### What are the relevant tickets?

- [CFG-2019](https://snyksec.atlassian.net/browse/CFG-2019)

### CODE REVIEW

**Comment by ofekatr** [comment]
Was there another PR opened for consuming the documentation URL on the CLI? 
Could you please link it in this PR's description?

File: src/cli/commands/test/iac/output.ts [hunk 1]
```diff
@@ -257,42 +199,22 @@ export function buildOutput({
...
-      const projects = results.length === 1 ? 'project' : 'projects';
-      summaryMessage +=
-        `\n\n\nTested ${results.length} ${projects}` +
-        summariseVulnerableResults(vulnerableResults, options) +
-        summariseErrorResults(errorResultsLength) +
-        '\n';
-    }
-  }
-
-  if (foundVulnerabilities) {
```
Comment (PeterSchafer) [inline src/cli/commands/test/iac/output.ts:272]: Question: This code block wan't related to isNewIacOutputSupported, is it still correct to remove it?

**Review by PeterSchafer — APPROVED** [review]

File: src/cli/commands/test/iac/output.ts (hunk 1 above)
Comment (ofekatr) [reply src/cli/commands/test/iac/output.ts:272]: > Question: This code block wan't related to isNewIacOutputSupported, is it still correct to remove it?

Good question, I've noticed the `summaryMessage` string this condition is for was only being populated for the legacy test output format. 
As such, I felt confident to remove it altogether, as well as the `else` condition linked to it.

File: test/jest/unit/lib/formatters/iac-output/text/user-messages.spec.ts [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    ${{}}                                       | ${true}
+    ${{ json: true }}                           | ${false}
+    ${{ sarif: true }}                          | ${false}
+    ${{ quiet: true }}                          | ${false}
+    ${{ json: true, sarif: true }}              | ${false}
+    ${{ json: true, quiet: true }}              | ${false}
+    ${{ sarif: true, quiet: true }}             | ${false}
+    ${{ json: true, sarif: true, quiet: true }} | ${false}
+  `(
+    'should return $expected, with: iacCliOutputFeatureFlag: $iacCliOutputFeatureFlag, options: $options',
```
Comment (ipapast) [inline test/jest/unit/lib/formatters/iac-output/text/user-messages.spec.ts:15]:
```suggestion
    'should return $expected, with options: $options',
```

**Review by ipapast — APPROVED** [review]

**Pull request merged**